### DISCUSSION TITLE
Add `--restart-container` option to `devspace sync` command

### DISCUSSION THREAD

**Discussion by petr-buchin** [devspace-sh/devspace#2389] [General]
**Is your feature request related to a problem?**  
No

**Which solution do you suggest?**  
It would be great to have ability to specify `--restart-container` option in `devspace sync` command, which would add functionality, already available via `restartContainer` option in `devspace.yaml` config.

**Which alternative solutions exist?**  
Alternative would be to use `devspace dev` pipeline

**Additional context**  
I need to run file sync separate from dev pipeline. DevSpace is a great tool, and it would be even better, if standalone `dev sync` command had the same functionality as `sync` section in `dev` pipeline. 

/kind feature

**Comment by FabianKramm** [comment]
Hey @petr-buchin! Thanks for creating this issue! I'm not sure if we should add this, since `devspace sync` is just meant to sync files into a container, but the `--restart-container` option would lead to replacing the container and injecting the restart helper. This is wanted in the regular dev workflow, but I'm not sure if we should add this for the sync command only as we would also need to add the other options for replacing a container or expect the restart helper to be injected already, which could cause confusion.

**Comment by alexandradragodan** [comment]
Thanks again for the suggestion, @petr-buchin . 
As Fabian mentioned, we're having some concerns related to the integration of the suggested feature in our overall product concept.

I have converted this issue to a discussion so we could address the topic further, if you'd like to. 
We would be happy to provide you with more details there!
